JPF/bad_buildings.sql

```diff
@@ -139,48 +175,64 @@ drop table if exists hs_permit_counts;
 ################
 create temporary table hs_permit_counts
 select 
-	bin_num,
-	CONCAT(trim(`house_num`)," ",trim(`street_name`)," ",trim(`borough`)) hs_dobp_address, 
+	bin_num bin,
+#	CONCAT(trim(`house_num`)," ",trim(`street_name`)," ",trim(`borough`)) address, 
+#   concat(trim(bl.boroid),trim(LPAD(dp.block, 5, '0')),trim(LPAD(dp.lot, 4, '0'))) bbl,
 	count(*) hs_permit_cnt
 from 
-	dob_permits 
+	dob_permits dp
+left join boro_lookup bl on bl.boro = dp.borough
 where 
 	work_type in ("OT","BL", "DM") 
 group by 
-	1,2;
-alter table hs_permit_counts add index(hs_dobp_address(255));
-alter table hs_permit_counts add index(bin_num);
+	1;
+#alter table hs_permit_counts add index(address(255));
+alter table hs_permit_counts add index(bin);
+#alter table hs_permit_counts add index(bbl);
 
 ################	
 drop table if exists permit_counts;
 ################
 create temporary table permit_counts
 select 
-	bin_num,
-	CONCAT(trim(`house_num`)," ",trim(`street_name`)," ",trim(`borough`)) dobp_address, 
+	bin_num bin,
+#	CONCAT(trim(`house_num`)," ",trim(`street_name`)," ",trim(`borough`)) address, 
+#	concat(trim(bl.boroid),trim(LPAD(dp.block, 5, '0')),trim(LPAD(dp.lot, 4, '0'))) bbl,
 	count(*) permit_cnt
 from 
-	dob_permits 
+	dob_permits dp
+left join boro_lookup bl on bl.boro = dp.borough
 group by 
-	1,2;
-alter table permit_counts add index(dobp_address(255));
-alter table permit_counts add index(bin_num);
+	1;
+#alter table permit_counts add index(address(255));
+alter table permit_counts add index(bin);
+#alter table permit_counts add index(bbl);
+
 
 ################
 drop table if exists violation_counts;
 ################
 create temporary table violation_counts
 select 
 	bin,
-	CONCAT(trim(`house_number`)," ",trim(`street`)," ",trim(bl.boro)) dobv_address, 
+	#CONCAT(trim(`house_number`)," ",trim(`street`)," ",trim(bl.boro)) address, 
+	#concat(trim(bl.boroid),trim(LPAD(block, 5, '0')),trim(LPAD(lot, 4, '0'))) bbl,
 	count(*) dob_violation_cnt
 from 
 	dob_violations dv
-left join boro_lookup bl on dv.boro = bl.boroid
+left join 
+	boro_lookup bl on dv.boro = bl.boroid
+where 
+	`violation_category` not like ("%dismissed%") and	`violation_category` not like ("%resolved%")
 group by 
-	1,2;
-alter table violation_counts add index(dobv_address(255));
+	1;
+#alter table violation_counts add index(address(255));
 alter table violation_counts add index(bin);
+#alter table violation_counts add index(bbl);
+#DELETE FROM violation_counts where bbl like "% %";
+#delete from violation_counts where bbl like "%N%";
+#delete from violation_counts where bbl regexp '[A-Za-z]';
+
 
 ################
 drop table if exists classA_vio_cnt;
```

Comment:
I can't comment on the below but all the same comments apply.

The class?_vio_cnt tables are especially good places for stored procedures, I think.